scripts/make_release_notes.py

```diff
@@ -0,0 +1,266 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Converts github flavored markdown changelogs to release notes.
+"""
+
+import argparse
+import re
+import subprocess
+
+
+PRODUCTS = {
+    'Firebase/Auth/CHANGELOG.md': '{{auth}}',
+    'Firebase/Core/CHANGELOG.md': '{{core}}',
+    'Firebase/Database/CHANGELOG.md': '{{database}}',
+    'Firebase/DynamicLinks/CHANGELOG.md': '{{ddls}}',
+    'Firebase/InAppMessaging/CHANGELOG.md': '{{inapp_messaging}}',
+    'Firebase/InstanceID/CHANGELOG.md': 'InstanceID',
+    'Firebase/Messaging/CHANGELOG.md': '{{messaging}}',
+    'Firebase/Storage/CHANGELOG.md': '{{storage}}',
+    'Firestore/CHANGELOG.md': '{{firestore}}',
+    'Functions/CHANGELOG.md': '{{cloud_functions}}',
+
+    # 'Firebase/CoreDiagnostics/CHANGELOG.md': '?',
+    # 'Firebase/InAppMessagingDisplay/CHANGELOG.md': '?',
+    # 'GoogleDataTransport/CHANGELOG.md': '?',
+    # 'GoogleDataTransportCCTSupport/CHANGELOG.md': '?',
+    # 'GoogleUtilities/CHANGELOG.md': '?',
+    # 'Interop/CoreDiagnostics/CHANGELOG.md': '?',
+}
+
+
+def main():
+  local_repo = find_local_repo()
+
+  parser = argparse.ArgumentParser(description='Create release notes.')
+  parser.add_argument('--repo', '-r', default=local_repo,
+                      help='Specify which GitHub repo is local.')
+  parser.add_argument('--only', metavar='VERSION',
+                      help='Convert only a specific version')
+  parser.add_argument('--all', action='store_true',
+                      help='Emits entries for all versions')
+  parser.add_argument('changelog',
+                      help='The CHANGELOG.md file to parse')
+  args = parser.parse_args()
+
+  if args.all:
+    text = read_file(args.changelog)
+  else:
+    text = read_changelog_section(args.changelog, args.only)
+
+  product = None
+  if not args.all:
+    product = PRODUCTS.get(args.changelog)
+
+  renderer = Renderer(args.repo, product)
+  translator = Translator(renderer)
+
+  result = translator.translate(text)
+  print(result)
+
+
+def find_local_repo():
+  url = subprocess.check_output(['git', 'config', '--get', 'remote.origin.url'])
+
+  # ssh or https style URL
+  m = re.match(r'^(?:git@github\.com:|https://github\.com/)(.*)\.git$', url)
+  if m:
+    return m.group(1)
+
+  raise LookupError('Can\'t figure local repo from remote URL %s' % url)
+
+
+class Renderer(object):
+
+  def __init__(self, local_repo, product):
+    self.local_repo = local_repo
+    self.product = product
+
+  def heading(self, heading):
+    if self.product:
+      return '### %s\n' % self.product
+
+    return heading
+
+  def bullet(self, spacing):
+    """Renders a bullet in a list.
+
+    All bulleted lists in devsite are '*' style.
+    """
+    return '%s* ' % spacing
+
+  def change_type(self, tag):
+    """Renders a change type tag as the appropriate double-braced macro.
+
+    That is "[fixed]" is rendered as "{{fixed}}".
+    """
+    return '{{%s}}' % tag
```

Comment:
`[added]` is rendered as `{{feature}}` in the official docs. See `googledata/devsite/site-firebase/en/_internal/includes/_elements.html?l=56`

Reply:
Done.

Confirmed with:

```
scripts/make_release_notes.py --only 4.1.4 Firebase/Database/CHANGELOG.md
### {{database}}

* {{feature}} Firebase Database is now community-supported on tvOS.
```

Grepping through our changelogs turns up these unique change type tags:

* added
* changed
* feature
* fixed
* removed

It looks like `[added]` is the only one that needs a mapping?

Reply:
Yeah, just `[added]`. It's unique because we use it commonly in our GitHub changelogs but it's not supported by devsite. Other elements can be added if we run into issues in the future.